### ISSUE TITLE
.NET 5 Preview 8 - dotnet build --no-incremental with analyzers enabled always fails

### ISSUE THREAD

**Issue by mareklinka** [dotnet/sdk#13360]
I just upgraded our production solution to .NET 5 Preview 8 as an experiment and immediately came across the following issue:

When running `dotnet build --no-increment` on the upgraded solution, I get several errors like this:
```
[MSBuild 16.8.0-preview-20414-02+a55ce4fbb]

...

CSC : error AD0001: Analyzer 'Microsoft.NetCore.Analyzers.Data.ReviewSqlQueriesForSecurityVulnerabilities' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [project path]
CSC : error AD0001: Analyzer 'Microsoft.NetCore.Analyzers.Runtime.DisposableFieldsShouldBeDisposed' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [project path]
...

```

When targeting 3.1, everything works fine:
```
[MSBuild 16.7.0+b89cb5fde]

...

Build succeeded.
    0 Warning(s)
    0 Error(s)
```

A very similar situation happens with normal (incremental) builds. With those, the solution eventually compiles but first I need to go through several failed incremental builds - each fails with different analyzer errors. This behavior is consistent - the failures are always the same for the given incremental build.

What is also interesting: when hitting `Rebuild` on the solution file in VS 2019.8 Preview 2.1 (the current preview), everything works fine every time.

If I disable build-time analyzers using `Directory.build.props`, build starts working again:
``` xml
<Project>
    <PropertyGroup>
        <RunAnalyzers>false</RunAnalyzers>
        <LangVersion>latest</LangVersion>
    </PropertyGroup>
</Project>
```

**Comment by mareklinka** [comment]
Update: it seems this might be an issue with MSBuild. When I try to build the main branch (netcoreapp3.1) using the 5 preview 8 SDK (set via `global.json`), I get the same analyzer exceptions.

``` json
{
    "sdk": {
        "version": "5.0.100-preview.8.20417.9"
    }
}
```

The build is then executed with the preview version of MSBuild shipping with VS 2019 Preview 2.1: `16.8.0-preview-20414-02+a55ce4fbb`

**Comment by mareklinka** [comment]
Looks like the problem was resolved by upgrading to VS 2019 Preview 3.0, MSBuild version 16.8.0-preview-20451-02+51a1071f8. Command line build now works as expected.
